### PR TITLE
Mount persist in installer, store and send logs

### PR DESCRIPTION
**Pull request by giggsoff**
Add support to store data in persist during installation. THis PR adds
logs to /persist/installer/installer.log and send them during first boot
 as logs to the controller.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

File: pkg/pillar/cmd/nodeagent/nodeagent.go [hunk 1]
```diff
@@ -613,6 +618,32 @@ func handleLastRebootReason(ctx *nodeagentContext) {
 	ctx.lastLock.Unlock()
 }
 
+// handleInstallationLog checks if first-boot file from installation exists
+// remove it and send log from installation to the controller
+func handleInstallationLog(_ *nodeagentContext) {
+	if fileExists(firstbootInstallFile) {
+		err := os.Remove(firstbootInstallFile)
```
Comment (eriknordmark) [inline pkg/pillar/cmd/nodeagent/nodeagent.go:625]: If we power off we'd potentiall miss some log entries. Better to do the remove at the end (means if power off we will repeat some logs which is better).

File: pkg/pillar/cmd/nodeagent/nodeagent.go [hunk 2]
```diff
@@ -613,6 +618,32 @@ func handleLastRebootReason(ctx *nodeagentContext) {
...
+		f, err := os.Open(installLog)
+		if err != nil {
+			log.Errorf("cannot open installation log: %s", err)
+			return
+		}
+		scanner := bufio.NewScanner(f)
+		buf := make([]byte, 0, maxReadSize)
+		scanner.Buffer(buf, maxReadSize)
+		for scanner.Scan() {
+			log.Noticeln(scanner.Text())
```
Comment (eriknordmark) [inline pkg/pillar/cmd/nodeagent/nodeagent.go:638]: Should we add/replace the source=nodeagent which log adds automatically with source=installer? I don't know if we need to do something special to replace i.e., avoid havig two source atttributes.

File: pkg/pillar/cmd/nodeagent/nodeagent.go (hunk 1 above)
Comment (giggsoff) [reply pkg/pillar/cmd/nodeagent/nodeagent.go]: I put small timeout before removing of the file to be on the safe side here.
Also I slightly reworked logic in device-steps to move the file from the installer (first-boot) into another file (send-require) to not assume twice that it was the first boot after installer.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Review by eriknordmark — APPROVED** [review]
Run eden once more

**Pull request merged**